go-controller/hybrid-overlay/pkg/controller/node_linux.go

```diff
@@ -29,56 +31,237 @@ const (
+	namespaceVTEP := namespace.GetAnnotations()[hotypes.HybridOverlayVTEP]
+	if !n.initialized {
+		node, err := n.wf.GetNode(n.nodeName)
 		if err != nil {
-			return fmt.Errorf("failed to add flows for pod %s/%s: %v", pod.Namespace, pod.Name, err)
+			return fmt.Errorf("hybrid overlay not initialized on %s, and failed to get node data: %v",
+				n.nodeName, err)
+		}
+		if err = n.ensureHybridOverlayBridge(node); err != nil {
+			return fmt.Errorf("failed to ensure hybrid overlay in pod handler: %v", err)
 		}
 	}
+	if n.drMAC == nil || n.drIP == nil {
+		return fmt.Errorf("empty values for DR MAC: %s or DR IP: %s on node %s", n.drMAC, n.drIP, n.nodeName)
+	}
+
+	for _, podIP := range podIPs {
+		var flows []string
+		cookie := podIPToCookie(podIP.IP)
+		if cookie == "" {
+			continue
+		}
+		// table 10 is pod dispatch - Incoming vxlan traffic towards pods
+		flows = append(flows, fmt.Sprintf(
+			"table=10,cookie=0x%s,priority=100,ip,nw_dst=%s,"+
+				"actions=set_field:%s->eth_src,set_field:%s->eth_dst,output:ext",
+			cookie, podIP.IP, n.drMAC.String(), podMAC))
+
+		if namespaceExternalGw == "" || namespaceVTEP == "" {
+			klog.Infof("Skipping hybrid overlay gw mode setup for pod %s, namespace does not have hybrid"+
+				"annotations, external gw: %s, VTEP: %s", pod.Name, namespaceExternalGw, namespaceVTEP)
+			n.updateFlowCacheEntry(cookie, flows, ignoreLearn)
+			continue
+		}
+
+		portMACRaw := strings.Replace(n.drMAC.String(), ":", "", -1)
+		vtepIPRaw := getIPAsHexString(net.ParseIP(namespaceVTEP))
+
+		// update map for tun to pod
+		n.tunMapMutex.Lock()
+		n.tunMap[podIP.IP.String()] = namespaceVTEP
+		// iterate and find all pods that belong to this VTEP and create learn actions
+		learnActions := ""
+		for pod, tun := range n.tunMap {
+			if tun == namespaceVTEP {
+				if len(learnActions) > 0 {
+					learnActions += ","
+				}
+				learnActions += fmt.Sprintf("learn("+
+					"table=20,cookie=0x%s,priority=50,"+
+					"dl_type=0x0800,nw_src=%s,"+
+					"load:NXM_NX_ARP_SHA[]->NXM_OF_ETH_DST[],"+
+					"load:0x%s->NXM_OF_ETH_SRC[],"+
+					"load:%d->NXM_NX_TUN_ID[0..31],"+
+					"load:0x%s->NXM_NX_TUN_IPV4_DST[],"+
+					"output:NXM_OF_IN_PORT[])",
+					podIPToCookie(net.ParseIP(pod)), pod, portMACRaw, hotypes.HybridOverlayVNI, vtepIPRaw)
+			}
+		}
+
+		// for arp request/response from vxlan, learn and add flow to table 20, for pod-> vxlan traffic
+		// special cookie needed here for tunnel
+		// tunnel cookie flows only contain a single flow ever, but it is updated by multiple pod adds
+		// so need proper locking around tunMap
+		// after learning actions, we need to resubmit the flow to the gw arp response table (2) so that we can respond
+		// back if this was an arp request
+		tunCookie := podIPToCookie(net.ParseIP(namespaceVTEP))
+		tunFlow := fmt.Sprintf("table=0,cookie=0x%s,priority=120,in_port=%s,arp,arp_spa=%s,tun_src=%s,"+
+			"actions=%s,resubmit(,2)",
```

Comment:
I tend to use "goto_table:2" rather than resubmit since I think that's clearer. Everywhere before this PR used goto FWIW.

Reply:
it doesn't really matter, because what ends up being in the flows is resubmit when you dump the table:
[root@ovn-control-plane ~]#  ovs-ofctl add-flow table=100,actions=goto_table:101
[root@ovn-control-plane ~]# ovs-ofctl dump-flows br-int |grep table=100
 cookie=0x0, duration=46.303s, table=100, n_packets=0, n_bytes=0, idle_age=46, actions=resubmit(,101)